### DISCUSSION TITLE
Notifications

### DISCUSSION THREAD

**Discussion by Doni49** [Nailik/rhasspy_mobile#333] [Ideas]
I would like the option to have Rhasspy throw a notification when Rhasspy is going to play TTS sound.  And another one when it's done playing the sounds.

My use case: I also have Tasker installed.  When Rhasspy is going to play TTS sounds (if the server sends text) then I want tasker to pause my music temporarily.  Tasker can see the notification and process pausing the music.

The second notification will allow tasker to know that it's time to resume the music.

**Comment by Nailik** [comment]
Maybe it would be more useful (for people not using tasker) to directly stop other media (music/videos) by the rhasspy app.
I think the AudioManager requestFocus function https://developer.android.com/reference/android/media/AudioManager#requestAudioFocus(android.media.AudioFocusRequest) should do the trick.

This could also help to stop media when recording (via touch or when wake word was detected).

I also already though about not recording (wake word) when Media is playing https://github.com/Nailik/rhasspy_mobile/issues/84

**Reply by Doni49** [reply]
As long as it can stop AND RESTART the music, that would work for this use case.

**Reply by Nailik** [reply]
jeah sure restarting is as important as stopping it

**Reply by Nailik** [reply]
@Doni49 i added a rudementary test implementation and created a pull request.
Once the workflow finished you should be able to download the debug build and verify it works for you as well https://github.com/Nailik/rhasspy_mobile/pull/336
I tested it with Android 13 and Spotify and Youtube (Premium) successfully.

When implementing the ui i will also add the option to mute completely or just make the music less loud (bring it to the back).

**Reply by Doni49** [reply]
"debug build"?  I'm currently using what I got from the Play store.  Will this be seen as a separate app on my device?

**Reply by Nailik** [reply]
No it will replace it, for testing purpose only. Make sure you backup your config before installing it.
I actually had to add sth to my workflows in order to build it.
You can find it here: https://github.com/Nailik/rhasspy_mobile/actions/runs/4682755427

**Reply by Doni49** [reply]
I don't have a way to compile it.  Do you have an APK available?

**Reply by Nailik** [reply]
Jeah in the "Artifacts" i linked. This is the direct download link: https://github.com/Nailik/rhasspy_mobile/suites/12196370799/artifacts/644270786

**Reply by Doni49** [reply]
I finally remembered to install it and test it.  So far it seems to work pretty well.  The only negative comment is that after I speak the wake word, it stops the music long enough to play the acknowledgement sound and then immediately resumes the music.

It makes me wonder if Rhasspy will understand the request if I ask while the music is playing.  Maybe it could wait to resume after it receives the request.

**Reply by Nailik** [reply]
Jeah it's currently only for playing audio, i will add it for recording audio as well when i make the final implementation with ui settings etc.
